### PR TITLE
Clean up a few warts with logging and presentation.

### PR DESCRIPTION
**Pull request by matttproud**
This commit drops our custom `Object#toString` mechanisms in favor
of Guava Objects#toStringHelper, which provides a sane and
stylistically consistent manner for doing this same task.  Multiline
emissions are deleted.

A few member fields are shortened to avoid violating style about
verbose names (think Zipf law).

Some of the fragile tests for stringification are deleted.  No sense
in testing something brittle.

Durations have been enhanced to use Joda Time's automatic humanization.

JvmFlagSet now uses an TreeMap<> instead of sorting on every
presentation.  Not that this behavior is performance critical, but might
as well save the work for an immutable type.

### CODE REVIEW

File: src/main/java/org/arbeitspferde/groningen/config/NamedConfigParamImpl.java [hunk 1]
```diff
@@ -55,7 +57,9 @@ public boolean equals(Object o) {
 
   @Override
   public String toString() {
-    return "@" + NamedConfigParam.class.getName() + "(value=" + value + ")";
+    return Objects.toStringHelper(NamedConfigParamImpl.class)
```
Comment (etheon) [inline src/main/java/org/arbeitspferde/groningen/config/NamedConfigParamImpl.java:60]: according to http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Objects.ToStringHelper.html, Objects.ToStringHelper is deprecated and will be removed in June2016. MoreObjects.ToStringHelper appears to be its replacement.

File: src/main/java/org/arbeitspferde/groningen/subject/SubjectGroup.java [hunk 2]
```diff
@@ -110,41 +111,21 @@ public String getClusterName() {
...
-    for (Subject subject : subjects) {
-      subjectIndices = commaJoiner.skipNulls().join(subjectIndices,
-          Integer.toString(subject.getSubjectIndex()));
-    }
-
-    return subjectIndices;
-  }
-
-  /** Get the number of subjects in the subject group we are allowed to control */
-  public int getSubjectCount() {
```
Comment (etheon) [inline src/main/java/org/arbeitspferde/groningen/subject/SubjectGroup.java:141]: Are you sure these weren't used elsewhere?

Comment (matttproud) [inline src/main/java/org/arbeitspferde/groningen/subject/SubjectGroup.java:141]: Compiler would have caught that.  If not, I can revert.

File: src/main/java/org/arbeitspferde/groningen/config/NamedConfigParamImpl.java (hunk 1 above)
Comment (matttproud) [inline src/main/java/org/arbeitspferde/groningen/config/NamedConfigParamImpl.java:60]: Great call.  We have a number of out-of-date patterns and library usages in the codebase that need to be corrected.

**Pull request merged**